### ISSUE TITLE
IPAPI Country Blocks

### ISSUE THREAD

**Issue by eurobank** [CIDRAM/CIDRAM#376]
Hi,
I have SG blocked with the CCS. I see today:

ID: 1673236194-166283-9655866217 »
Date/Time: Mon, 09 Jan 2023 05:49:54 +0200 »
IP address: 207.148.117.235 »
User agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0 »
Signatures count: 1 »
Signatures reference: Core.php:L2129 »
Why blocked: vultrusercontent.com! »
Reconstructed URI: https : // *****.***** /e/data/js/ajax.js »
Infractions: 1 »
Request method: GET »
Hostname: 207.148.117.235.vultrusercontent.com »

But https://ipinfo.io/AS20473/207.148.116.0/22

is Singapore and was blocked due to my Aux Rules (*vultusercontent.com*) and not from the IPAPI.

**Comment by Maikuolan** [comment]
At my end, it seems to be correctly identifying the IP address as originating from Singapore and blocking accordingly.

![Untitled](https://user-images.githubusercontent.com/12571108/211278019-5cfcdcae-b637-40ad-bed9-03c94915c814.png)

Would be great to know whether it's identifying correctly at your end to help debug further, but I'd normally suggest checking the cache page to confirm that, and given that the cache page isn't working as intended at the moment with Memcached.. becomes a little tricky to do.

Need to sort out a few offline things right now but will think on the problem for a bit and try to reply back a little later today.

**Comment by eurobank** [comment]
Second issue:

ID: 1673249005-440368-0929962024 »
Date/Time: Mon, 09 Jan 2023 09:23:25 +0200 »
IP address: 128.199.78.77 »
User agent: python-requests/2.28.1 »
Signatures count: 1 »
Signatures reference: Core.php:L2129 »
Why blocked: python! »
Reconstructed URI: **************
Infractions: 4 »
Request method: GET »
Hostname: - »

https://www.abuseipdb.com/check/128.199.78.77

Singapore. But blocked from Aux Rules (*python* user agent)


![Screenshot](https://user-images.githubusercontent.com/4467158/211279672-55894004-44a3-43a7-ae30-51a3eea94c3d.png)

**Comment by eurobank** [comment]
It seems to work BUT not all the time:

ID: 1673241950-312651-5802246249 »
Date/Time: Mon, 09 Jan 2023 07:25:50 +0200 »
IP address: 184.168.98.87 »
User agent: - »
Signatures count: 1 »
Signatures reference: 184.168.98.0/24 »
Why blocked: CC (IP-API, "SG")! »
Reconstructed URI: ********
Infractions: 1 »
Request method: GET »
Hostname: 87.98.168.184.host.secureserver.net »

**Comment by eurobank** [comment]
And again it works now but hard to tell when it doesn't:

ID: 1673260710-278116-5767696140 »
Date/Time: Mon, 09 Jan 2023 12:38:30 +0200 »
IP address: 207.148.117.235 »
User agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0 »
Signatures count: 2 »
Signatures reference: 207.148.117.0/24, Core.php:L2129 »
Why blocked: CC (IP-API, "SG"), vultrusercontent.com! »
Reconstructed URI: ***************
Infractions: 3 »
Request method: GET »
Hostname: 207.148.117.235.vultrusercontent.com »

ID: 1673260707-955462-3445045714 »
Date/Time: Mon, 09 Jan 2023 12:38:27 +0200 »
IP address: 45.77.168.140 »
User agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0 »
Signatures count: 2 »
Signatures reference: 45.77.168.0/24, Core.php:L2129 »
Why blocked: CC (IP-API, "SG"), vultrusercontent.com! »
Reconstructed URI: ************************
Infractions: 2 »
Request method: GET »
Hostname: 45.77.168.140.vultrusercontent.com »

**Comment by eurobank** [comment]
@Maikuolan Could it be that IF an Aux Rule is trigered, then NO check is done (or none is reported/logged) from IPAPI?

EDIT: Because my first posts, ALL have Aux Rules triggered.

**Comment by 737simpilot** [comment]
@Maikuolan Are country codes supposed to be line by line? Does it matter?

**Comment by Maikuolan** [comment]
> @Maikuolan Are country codes supposed to be line by line? Does it matter?

For v1+v2, they needed to be separated by commas (because v1+v2 didn't support multiline values).

For v3 (up until very recently), they needed to be separated by line breaks (i.e., one country code per line), because v3 supports multiline values, and accordingly, when rewriting the modules for v3, I'd opted to take advantage of the newly available multiline support.

However, I pushed a commit to v3 on January 3rd which now allows any kind of space character or comma to be used (commas, line breaks, tabs, spaces, whatever), so it shouldn't matter anymore.

**Comment by 737simpilot** [comment]
👍
